tests/integration-tests/tests/networking/test_networking.py

```diff
@@ -0,0 +1,188 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import datetime
+import os
+
+import boto3
+import pytest
+
+from assertpy import assert_that
+from cfn_stacks_factory import CfnStack, CfnStacksFactory
+from conftest import _AVAILABILITY_ZONE_OVERRIDES
+
+TIMESTAMP = "-{:%Y%m%d%H%M%S}".format(datetime.datetime.utcnow())
+
+
+@pytest.fixture()
+def networking_stack_factory(request):
+    """Define a fixture to manage the creation and destruction of CloudFormation stacks."""
+    factory = CfnStacksFactory()
+    yield factory
+    if not request.config.getoption("no_delete"):
+        factory.delete_all_stacks()
+
+
+def get_internet_gateway_id(ec2_client, vpc_id):
+    internet_gateway_response = ec2_client.describe_internet_gateways(
+        Filters=[{"Name": "attachment.vpc-id", "Values": [vpc_id]}]
+    )
+    return (
+        internet_gateway_response["InternetGateways"][0]["InternetGatewayId"]
+        if internet_gateway_response["InternetGateways"]
+        else ""
+    )
+
+
+def get_cfn_parameters(availability_zone, internet_gateway_id, public_cidr, vpc_id, private_cidr=None):
+    """Create cloudformation-compatible stack parameter given the variables."""
+    parameters = [
+        {"ParameterKey": "AvailabilityZone", "ParameterValue": availability_zone},
+        {"ParameterKey": "InternetGatewayId", "ParameterValue": internet_gateway_id},
+        {"ParameterKey": "PublicCIDR", "ParameterValue": public_cidr},
+        {"ParameterKey": "VpcId", "ParameterValue": vpc_id},
+    ]
+    if private_cidr:
+        parameters.append({"ParameterKey": "PrivateCIDR", "ParameterValue": private_cidr})
+    return parameters
+
+
+def load_public_template():
+    with open(os.path.join("..", "..", "cloudformation", "networking_configuration", "public.cfn.json")) as cfn_file:
+        file_content = cfn_file.read()
+    return file_content
+
+
+def load_public_private_template():
+    with open(
+        os.path.join("..", "..", "cloudformation", "networking_configuration", "public-private.cfn.json")
+    ) as cfn_file:
+        file_content = cfn_file.read()
+    return file_content
+
+
+def test_public_configuration_no_internet_gateway(region, vpc_stacks, networking_stack_factory):
+    ec2_client = ec2_conn(region)
+    vpc_id = vpc_stacks[region].cfn_outputs["VpcId"]
+    public_subnet_cidr = "10.0.3.0/24"
+    availability_zone = _AVAILABILITY_ZONE_OVERRIDES.get(region, "")
+    internet_gateway_id = get_internet_gateway_id(ec2_client, vpc_id)
```

Comment:
I would add this code to CfnStack and read all the resources from the Stacks Resources whenever possible. This reduces boto3 calls and also requires less IAM policies

```
@property
    def cfn_resources(self):
        """
        Return the CloudFormation stack resources for the cluster.
        Resources are retrieved only once and then cached.
        """
        if not self.__cfn_resources:
            self.__cfn_resources = retrieve_cfn_resources(self.cfn_name, self.region)
        return self.__cfn_resources
```